core/modules/node/node.entity.inc

```diff
@@ -206,6 +216,172 @@ class Node extends Entity {
     );
   }
 
+  /**
+   * Overrides Entity::createAccess().
+   */
+  public static function createAccess($bundle = NULL, $account = NULL) {
+    $rights = &backdrop_static(__METHOD__, array());
+
+    // If no user object is supplied, the access check is for the current user.
+    if (empty($account)) {
+      $account = $GLOBALS['user'];
+    }
+
+    // If we've already checked access for this node, user and op, return from
+    // cache.
+    if (isset($rights[$account->uid][$bundle])) {
+      return $rights[$account->uid][$bundle];
+    }
+
+    if (user_access('bypass node access', $account)) {
+      $rights[$account->uid][$bundle] = TRUE;
+      return $rights[$account->uid][$bundle];
+    }
+    if (!user_access('access content', $account)) {
+      $rights[$account->uid][$bundle] = FALSE;
+      return $rights[$account->uid][$bundle];
+    }
+
+    // We grant access to the node if both of the following conditions are met:
+    // - No modules say to deny access.
+    // - At least one module says to grant access.
+    // If no module specified either allow or deny, we fall back to the
+    // node_access table.
+    $access = module_invoke_all('node_access', $bundle, 'create', $account);
+    if (in_array(NODE_ACCESS_DENY, $access, TRUE)) {
+      $rights[$account->uid][$bundle] = FALSE;
+      return $rights[$account->uid][$bundle];
+    }
+    elseif (in_array(NODE_ACCESS_ALLOW, $access, TRUE)) {
+      $rights[$account->uid][$bundle] = TRUE;
+      return $rights[$account->uid][$bundle];
+    }
+
+    $rights[$account->uid][$bundle] = FALSE;
+    return $rights[$account->uid][$bundle];
+  }
+
+  /**
+   * Overrides Entity::access().
+   *
+   * @param string $op
+   *   The operation to be performed on the node. Possible values are:
+   *   - create
+   *   - view
+   *   - update
+   *   - delete
+   * @param User|AnonymousUser|object $account
+   *   (optional) The user to check for. Leave it to NULL to check for the
+   *   global user.
+   *
+   * @return bool
+   *   TRUE if access is granted, FALSE otherwise.
+   */
+  public function access($op, $account = NULL) {
+    // Casting class with private property causes errors due to added prefix.
+    // e.g. "\0" . 'Node' . "\0". So use static array instead.
+    $rights = &backdrop_static(__METHOD__, array());
```

Comment:
Looks like this would create a static just with the name `access` because that is the value of `__METHOD__`, which isn't very identifying. If needed just use a hard-coded string like `'node_access'`, matching what it was before.

I'm also not understanding why we have a `private $rights` property defined above, but then we're using a static here. Should the property be removed?

Reply:
Should this be updating the private $rights?

Reply:
ok.